index.js

```diff
@@ -16,6 +16,31 @@ function readUInt64LE(buf, offset = 0) {
 	return n;
 }
 
+const tarHeaderChecksumMatches = buf => { // Does not check if checksum field was valid
+	if (buf.length < 512) {
+		return false;
+	}
+
+	let sum = 256;
+	let signedBitCount = 0;
+
+	for (let i = 0; i < 148; i++) {
+		const byte = buf[i];
+		sum += byte;
+		signedBitCount += byte & 0x80;
+	}
+
+	for (let i = 156; i < 512; i++) {
+		const byte = buf[i];
+		sum += byte;
+		signedBitCount += byte & 0x80;
+	}
+
+	const readSum = parseInt(buf.toString('utf8', 148, 154), 8);
+
+	return sum === readSum || ((sum - (signedBitCount << 1)) === readSum);
+};
```

Comment:
Can you add some comments in this function to explain what's going on?

Reply:
done